Presentations/Week17/expressions/app.R

```diff
@@ -41,21 +44,42 @@ ui <- fluidPage(
 )
 
 
-server <- function(input, output) {
+server <- function(input, output, session) {
+  
+  # grab the formals based on the user specified function
+  all_formals <- reactive({ get_all_formals(!!input$func) })
+  
+  observeEvent(input$func, {
+    x <- ifelse(input$func == "", 0, length(formals()$formals))
+    updateNumericInput(session, inputId = "n", value = x)
+  })
+
+  # create a list of formals that are not elipse
+  # then TRUE FALSE if it contains an elipse
+  formals <- reactive({
+    req(all_formals())
+    list(
+      formals = all_formals()[all_formals() != "..."],
+      elipse = "..." %in% all_formals()
+    )
+  })
```

Comment:
I think you mean dots here, or maybe ellipse (two Ls 😝 ) instead of ellipse :) dots probably more consistent with what it's called in R as well